### DISCUSSION TITLE
LCOM and DFASL files

### DISCUSSION THREAD

**Discussion by blakemcbride** [Interlisp/medley#114] [Q&A]
Greetings,

I believe that DFASL and LCOM files are compiled lisp code.  (If not, please help clarify.)  My questions are:  

1. What is the difference between each?

2. How are each created?

3. How are each loaded?

Thanks!

Blake

**Comment by masinter** [comment]
the question of the differences between them isn't so simple. _Some_ differences are:
* DFASL files are made by cl:compile-file, while .LCOM files are produced by IL:TCOMPL and IL:BCOMPL
* I've heard but haven't validated that only cl:compile-file can compile common lisp code (what happens if you try?)
* I can see that IL:LOAD is different from CL:LOAD but it seems like IL:LOAD works on either
* I've heard that IL:LOADFN for loading a single compiled function doesn't work with DFASL files
* They produce different code

Why there are two and never converged is a mystery that this project should determine -- maybe there is some explanation in the Medley user's guide or release notes or something
